presto-matching/src/main/java/com/facebook/presto/matching/UsageCallSite.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.matching;
+
+import com.google.common.base.Splitter;
+
+import static com.google.common.collect.Iterables.getLast;
+import static java.lang.String.format;
+import static java.util.Arrays.stream;
+
+public class UsageCallSite
+{
+    private final String className;
+    private final String methodName;
+    private final String fileName;
+    private final int lineNumber;
+
+    public static UsageCallSite get()
+    {
+        StackTraceElement[] stackTrace = new Throwable().getStackTrace();
+        StackTraceElement callSite = stream(stackTrace)
+                .filter(element -> !packageName(element).equals("com.facebook.presto.matching"))
+                .findFirst()
+                .get();
```

Comment:
That's the way logging frameworks do that, BTW. (Logback, for example). I'll gladly answer your questions and and arguments if you provide them.